tracer/src/Datadog.Trace/Debugger/Helpers/StringExtensions.cs

```diff
@@ -15,16 +15,16 @@ internal static class StringExtensions
         public static string ToUUID(this string input)
         {
             byte[] hash;
-            using (var md5 = MD5.Create())
+            using (var sha256 = SHA256.Create())
             {
-                hash = md5.ComputeHash(Encoding.UTF8.GetBytes(input));
+                hash = sha256.ComputeHash(Encoding.UTF8.GetBytes(input));
             }
 
             hash[6] &= 0x0f;
             hash[6] |= 0x30;
             hash[8] &= 0x3f;
             hash[8] |= 0x80;
-            var hex = BitConverter.ToString(hash).Replace("-", string.Empty).ToLower();
+            var hex = BitConverter.ToString(hash).Replace("-", string.Empty).Substring(0, 32).ToLower();
             return hex.Insert(8, "-").Insert(13, "-").Insert(18, "-").Insert(23, "-");
```

Comment:
I am trying to think about a solution for this. If it has to be MD5 the algorithm used, maybe, an additional check, when checking the stack before generating the span, could check if the call is coming from the StringExtensions class of DD. In 2 or 3 months, we will have the callsite instrumentation and this particular problem will be gone because we will exclude from instrumentation all the DD code, but in the meantime, it might be a solution. What do you think?

Reply:
I have left it as it was, with the md5. I check later, during the span generation, the origin of the call.